src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -344,141 +354,154 @@ private static unsafe bool AllowHyphenDuringParsing(NumberFormatInfo info)
                        };
             }
 
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            private static unsafe char* MatchNegativeSignChars(char* p, char* pEnd, string negativeSign, bool allowHyphenDuringParsing)
+            private static bool IsDecimalChar(char ch)
             {
-                char *ret = MatchChars(p, pEnd, negativeSign);
-                if (ret == null && allowHyphenDuringParsing && p < pEnd && *p == '-')
-                {
-                    ret = p + 1;
-                }
+                return (ch >= '0') && (ch <= '9');
+            }
 
-                return ret;
+            private static bool IsHexadecimalChar(char ch)
+            {
+                return IsDecimalChar(ch)
+                    || ((ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F'));
             }
 
-            private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles options, ref NumberBuffer number, StringBuilder? sb, NumberFormatInfo numfmt, bool parseDecimal)
```

Comment:
Just to confirm (because the GH diff here is terrible): No changes were made to this function right?

Reply:
Some methods was extracted from this method, because it was too large and it was renamed to TryParseNumber.